### PR TITLE
kitchen-sync 0.38 (new formula)

### PR DESCRIPTION
**Pull request by denji**
- 0.38, OpenSSL removed from the CMakeList for OS X.

### CODE REVIEW

File: Library/Formula/kitchen-sync.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    assert File.exist?("#{bin}/ks")
+    assert File.exist?("#{bin}/ks_mysql") if build.with? "mysql"
+    assert File.exist?("#{bin}/ks_postgresql") if build.with? "postgresql"
```
Comment (bfontaine) [inline Library/Formula/kitchen-sync.rb]: Could you add a more substantial test than e.g. asserting files existence or `--help`/`--version`? Thanks!

Comment (denji) [inline Library/Formula/kitchen-sync.rb]: @bfontaine There is no argument for `-h (--help)`, we can connect to the database you want to synchronize or run the tests work with base.

Comment (bfontaine) [inline Library/Formula/kitchen-sync.rb]: Mmmh I see, there’s no way to test this without a local database, and no `--help`/`--version` flag. @mikemcquaid Thoughts?

File: Library/Formula/kitchen-sync.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  desc "Fast efficiently sync database without dumping & reloading"
+  homepage "https://github.com/willbryant/kitchen_sync"
+  url "https://github.com/willbryant/kitchen_sync/archive/0.37.tar.gz"
+  sha256 "a78206cdd3ac4d4aa646cdfc8ae34c5df87179223c59d5dcca625760c68ae4e7"
+  head "https://github.com/willbryant/kitchen_sync.git"
+
+  depends_on "cmake" => :build
+  depends_on "boost"
+  depends_on :mysql => :recommended
+  depends_on :postgresql => :optional
```
Comment (bfontaine) [inline Library/Formula/kitchen-sync.rb]: It should also `depends_on "openssl"`, as mentioned in the [upstream docs](https://github.com/willbryant/kitchen_sync/blob/master/INSTALL.md#installing-kitchen-sync).

File: Library/Formula/kitchen-sync.rb (hunk 1 above)
Comment (willbryant) [inline Library/Formula/kitchen-sync.rb]: Hey guys, great to see this.  I believe you can remove the openssl dependency, that was my bad - I forgot to update the installation instructions when I added direct support for Common Crypto on OS X, so now OpenSSL is only needed on Linux and so on.

Let me know if it isn't compiling without it or something - unfortunately all my machines have OpenSSL already so hard for me to test this properly.  All I know for sure is that we don't get the deprecation warnings...

**Comment by bfontaine** [comment]
The test is failing because of an issue in the core that has since been resolved, could you please rebase on top of `master`?

Also, regarding the OpenSSL issue, if you don’t have Homebrew’s OpenSSL it builds against OSX’ outdated one:

```
==> Downloading https://github.com/willbryant/kitchen_sync/archive/0.37.tar.gz
Already downloaded: /Library/Caches/Homebrew/kitchen-sync-0.37.tar.gz
==> cmake . -DCMAKE_C_FLAGS_RELEASE= -DCMAKE_CXX_FLAGS_RELEASE= -DCMAKE_INSTALL_PREFIX=/usr/local/Cel
==> make install
Warning: object files were linked against system openssl
These object files were linked against the deprecated system OpenSSL.
Adding `depends_on "openssl"` to the formula may help.
  /usr/local/Cellar/kitchen-sync/0.37/bin/ks_mysql
==> Summary
🍺  /usr/local/Cellar/kitchen-sync/0.37: 43 files, 1,6M, built in 29 seconds
```

**Comment by willbryant** [comment]
Hmm, that isn't meant to happen... we need to skip the `find_package(OpenSSL REQUIRED)` on OS X I guess?

**Comment by willbryant** [comment]
Pushed 0.38 with OpenSSL removed from the CMakeList for OS X.

**Comment by denji** [comment]
@willbryant http://bot.brew.sh/job/Homebrew%20Pull%20Requests/27254/version=mountain_lion/

**Comment by DomT4** [comment]
Looks like it needs `cxx11` probably.

**Comment by denji** [comment]
@DomT4 Now it works fine!

**Comment by mikemcquaid** [comment]
Thanks @denji!

**Comment by mikemcquaid** [comment]
Merged in 19d2554.

**Pull request closed without merging**